packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -186,7 +186,7 @@ public void onCancel(Object o) {
           }
         });
 
-    surface = new Surface(textureEntry.surfaceTexture());
+    surface = surfaceProducer.getSurface();
```

Comment:
Note, the returned Surface may become invalid after a suspend / resume. Is there a place in the plugin that will be notified when we resume? We should update the video surface in that callback.